addons/stock/models/stock_move_line.py

```diff
@@ -485,7 +485,10 @@ def unlink(self):
             if ml.product_id.type == 'product' and ml.move_id and not ml.move_id._should_bypass_reservation(ml.location_id) and not float_is_zero(ml.product_qty, precision_digits=precision):
                 self.env['stock.quant']._update_reserved_quantity(ml.product_id, ml.location_id, -ml.product_qty, lot_id=ml.lot_id, package_id=ml.package_id, owner_id=ml.owner_id, strict=True)
         moves = self.mapped('move_id')
+        package_levels = self.package_level_id.filtered(lambda pl: len(pl.move_line_ids) == 1 and not pl.move_ids)
```

Comment:
can't you unlink the package levels directly here ?

Reply:
Unlinking the package level also triggers unliking the move line which would lead to infinite recursion 
That's why I added the check

Reply:
ok, thank you. Anyway, I would be more generic and checking all stock move line of the package are deleted to delete it. `len(p1.move_line_ids) == 1` seems a little bit too restricted to me